src/rewards/FlywheelGaugeRewards.sol

```diff
@@ -86,84 +103,165 @@ contract FlywheelGaugeRewards is Auth, IFlywheelRewards {
     /**
         @notice Iterates over all live gauges and queues up the rewards for the cycle
         @return totalQueuedForCycle the max amount of rewards to be distributed over the cycle
-        @dev critical that there is no pagination to prevent double spending. All gauges must be synced atomically, unless the gaugeToken checkpoints rewards.
 
         GAS: Average path for `n` gauges - SLOAD * (n + 7) + SSTORE * (n + 1) + 1 token transfer
     */
     function queueRewardsForCycle() external requiresAuth returns (uint256 totalQueuedForCycle) {
+        // next cycle is always the next even divisor of the cycle length above current block timestamp.
+        uint32 currentCycle = uint32(block.timestamp) / gaugeCycleLength * gaugeCycleLength;
```

Comment:
you could memoize `gaugeCycleLength` to avoid reading twice

Reply:
its immutable so it gets compiled into the runtime code and there is no SLOAD